frontend/gist-frontend/src/Components/CreateUserComponent.tsx

```diff
@@ -9,6 +10,30 @@ interface ICreateUserComponentState {
   imageName: string;
 }
 
+const CreateUserSchema = Yup.object().shape({
+  fullName: Yup.string().required("required"),
+  username: Yup.string() // restrictions on username?
+    .required("required")
+    .min(8, "too short!")
+    .max(25, "too long!"),
+  password: Yup.string()
+    .required("required")
+    .matches(
+      /^(?=.*?[A-Z])(?=.*?[a-z])(?=.*?[0-9])(?=.*?[#?!@$%^&*-]).{8,}$/,
+      "password must have at least one lowercase, one uppercase, one numeric, one special character, and be at least 8 characters in length"
+    ),
+  password2: Yup.string()
```

Comment:
does password2 match password?

Reply:
Yes? I don't know what you mean here. I know I could make this simpler and don't need to use the same regex as the password before it but haven't changed it yet. I can probably do something like if password === secondPassword then validate.

Reply:
ya i meant like does this check that the password is the same as the second one or does it check that they follow the same rules?

Reply:
I'll just check that it's the same, because of it's the same then it obviously follows the same rules.